### ISSUE TITLE
Button renders wrong after mouse leave

### ISSUE THREAD

**Issue by jorisvergeer** [dotnet/wpf#707]
* .NET Core Version: 3.0 preview 5
* Windows version: 1803 build 17134.765
* Does the bug reproduce also in WPF for .NET Framework 4.8?: ~I don't know.~ It works fine on 4.6/4.7 and 4.8

 **Problem description:**
With this XAML:
```
<Window x:Class="RetailEntertainment.Zorgstand.SignIn.Desktop.Views.AddNewConnectionWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:local="clr-namespace:RetailEntertainment.Zorgstand.SignIn.Desktop.Views"
        xmlns:vms="clr-namespace:RetailEntertainment.Zorgstand.SignIn.Desktop.ViewModels"
        mc:Ignorable="d"
        d:DataContext="{d:DesignInstance vms:AddNewRegistrationDialogViewModel}"
        Title="New registration" Width="400" Height="100">
    <StackPanel>
        <TextBlock Text="Entert the connection string received from the portal:" />
        <TextBox Text="{Binding ConnectionString}" />
        <StackPanel Orientation="Horizontal" HorizontalAlignment="Right">
            <Button Content="Cancel" Margin="4" Padding="2" Command="{Binding CloseDialogCommand}"/>
            <Button Content="Save" Margin="4" Padding="2" Command="{Binding TestAndAddRegistrationCommand}"/>
        </StackPanel>
    </StackPanel>
</Window>
```
The buttons render wrong after mouse leaves.
Before:
![image](https://user-images.githubusercontent.com/1906639/57971363-4e266700-798d-11e9-8fb7-61618ae0e269.png)
After:
![image](https://user-images.githubusercontent.com/1906639/57971367-567ea200-798d-11e9-890f-3980dcac4fc0.png)
 
 **Actual behavior:** 
It looks like the button renders as if it has its parent size. 
After resizing the window, it renders normally again.
 
 **Expected behavior:**
No weird rendering
 
 **Minimal repro:**
 Code is above

**Comment by lindexi** [comment]
I run your code but work fine.

**Comment by jorisvergeer** [comment]
[App.xaml.zip](https://github.com/dotnet/wpf/files/3196758/App.xaml.zip)
I have this issue with `dotnet new wpf` and just showing my window.

![image](https://user-images.githubusercontent.com/1906639/58004474-95bf0700-7ae3-11e9-8dd2-4e31e92a6812.png)

**Comment by jorisvergeer** [comment]
Let me add some information:
My laptop had an Intel i7 with integrated graphics and a Nvidia 1050 GPU.

To me it looks like that during drawing some framebuffers or textures are messed up.

**Comment by jorisvergeer** [comment]
I tried this on another PC at work.  There it shows the correct behavior.

Throwing this bug in the category that depends on the environment it runs on.

**Comment by lindexi** [comment]
@jorisvergeer Could you find if some drivers are not installed on your device?

**Comment by jorisvergeer** [comment]
As far as I am aware, I have all drivers installed. I was able to play some games last night.

**Comment by jorisvergeer** [comment]
Are there any commands or tools I can run on my PC to diagnose the drawing?

**Comment by jorisvergeer** [comment]
Anyway, 
What I guess that happend om my machine it this:

(Approximation with Paint)

It wants to draw this:
![image](https://user-images.githubusercontent.com/1906639/58009993-da509f80-7aef-11e9-9175-6e370e4bc872.png)
Accidently draws a button on with its parents size (I saw something like this on one occasion when switching windows):
![image](https://user-images.githubusercontent.com/1906639/58010050-f05e6000-7aef-11e9-9ba3-6caba496e0d8.png)
Then only draws the portion in the buttons region:
![image](https://user-images.githubusercontent.com/1906639/58010162-2ac7fd00-7af0-11e9-9d36-c86ffdd15506.png)

**Comment by lindexi** [comment]
> Are there any commands or tools I can run on my PC to diagnose the drawing?
> Or to debug the whole .net wpf drawing code.

Yes, the [Performance Profiling Tools for Windows Presentation Foundation](https://download.microsoft.com/download/A/6/A/A6AC035D-DA3F-4F0C-ADA4-37C8E5D34E3D/setup/WinSDKPerformanceToolKit_amd64/wpt_x64.msi) may help you to find the dirty region.

**Comment by jorisvergeer** [comment]
@lindexi That tool does not seem to work with dotnet core 3.0 WPF

**Comment by walterlv** [comment]
@jorisvergeer The tool works fine with .NET Core 3.0 WPF application. I've tested it a few weeks ago.

I wrote a post for my demo and you may read the English translated version:

- [Optimize rendering performance by analyzing WPF's rendered dirty regions - walterlv](https://translate.google.com/translate?sl=zh-CN&tl=en&u=https%3A%2F%2Fblog.walterlv.com%2Fpost%2Fwpf-rendering-dirty-region.html)

**Comment by jorisvergeer** [comment]
@walterlv @lindexi You are right. But you have to have the "Performator" tab open. Otherwise it doesn't work.

**Comment by 0xd4d** [comment]
Possibly related to https://github.com/dotnet/wpf/issues/319

**Comment by jorisvergeer** [comment]
@0xd4d Its probably related.

**Comment by jorisvergeer** [comment]
I don't know if it is worth anything:
- If I tick "Disable dirty region support" or "Show dirty-region update overlay", then it renders just fine.
- If I tick "Clear back-buffer before rendering" and then hover over a button, then the content of the window turns black, except for the button which renders correctly.
![image](https://user-images.githubusercontent.com/1906639/58040675-49e97d80-7b36-11e9-8c09-d199123bcfaf.png)
- Any other checkdox has no noticable effect

**Comment by jorisvergeer** [comment]
Just captured another fun example:
![image](https://user-images.githubusercontent.com/1906639/58040829-bebcb780-7b36-11e9-993e-90c47ab8fb39.png)

**Comment by jorisvergeer** [comment]
I guess that somewhere, somehow, some active framebuffers are being reused, leaked or mixed up.

**Comment by stevenbrix** [comment]
@jorisvergeer what's your screen resolution and scale factor? Are you using multiple monitors?

**Comment by jorisvergeer** [comment]
@stevenbrix My laptop has 1920x1080 as resulution is uses the default scale factor. 
My laotp is of type MSI GF63 8RD

**Comment by weltkante** [comment]
I doubt that its got anything to do with DPI/resolution, that would probably keep aspect ratio and not just stretch the image to the window size. Probably more likely something to do with drivers/hardware. The other issue linked above #319 (closed as duplicate) was also reproduced on an integrated intel chipset.

Doesn't WPF have a way to force software rendering? Goole finds things like [this](https://stackoverflow.com/questions/4951058/software-rendering-mode-wpf), if these work in .NET Core maybe someone who can reproduce the problem can check if it goes away when forcing WPF to render in software.

**Comment by MrPurple6411** [comment]
Just as a bit of extra info... I am running a MSI GL73 8RC so pretty close to OP's computer and I am having this same issue using version 2.2.401 skd and 2.2.6 runtime.   I think it might be a graphics driver issue but at the same time it happens when I force it to run on the gtx 1050 as well as the intel integrated chip.   All drivers are current and up to date.

**Comment by stevenbrix** [comment]
> I am having this same issue using version 2.2.401 skd and 2.2.6 runtime

@MrPurple6411 WPF has only been ported to .NET Core beginning with 3.0, so I'm curious how you are using 2.x sdk and runtime?

My Surface Book 2 has the gtx 1060 and integrated intel graphics card and i'm not seeing this, so I wonder if it has something to do with the 1050. I'll be working on getting a device to see if i can repro this. Thanks everyone for the info so far!

**Comment by stevenbrix** [comment]
@PascalKoe (who opened #319) mentioned that he was just using the integrated Intel graphics. I'd be curious to know more about your hardware. I tried this on a Lenova Yoga with the same Intel integrated graphics and Nvidia GTX1050 configuration and didn't reproduce this. I'm beginning to think it might not have as much to do with the GTX1050 as it does with the intel integrated graphics. I'm hoping I can get my hands on one of these MSI laptops to reproduce this issue.

For those that can reproduce this, are there any other scenarios or times where you see this behavior? Does this always happen with the sample app and DnSpy, or is there a time when it renders as expected?

If you could try and build the D3D9 sample [here](https://github.com/microsoft/Windows-classic-samples/tree/master/Samples/Win7Samples/multimedia/direct3d/direct3d9ex) and see if there are any issues with it. You'll probably also need to install the DirectX SDK [here](https://www.microsoft.com/en-us/download/details.aspx?id=6812). This will help us understand if it's an issue with WPF or the graphics stack.

**Comment by stevenbrix** [comment]
@jorisvergeer would you be able to share a VS Live Share session with me sometime so I can take a look under a debugger?

**Comment by jorisvergeer** [comment]
@stevenbrix That would be possible

**Comment by stevenbrix** [comment]
@jorisvergeer actually even better would be to collect a Time Travel trace using WinDbg Preview and share that with us.

### Steps for collecting a Time Travel trace
1. If you don't already have WinDbg Preview, you can install it from here: https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/windbg-install-preview
2. Launch WinDbg Preview as Admin
3. Go to File ->Start debugging -> Launch executable (advanced)
4. Give path to executable and check "Record process with Time Travel Debugging"

I've included a picture below of what your window should look like:

![image](https://user-images.githubusercontent.com/9649518/63202479-ad545000-c03e-11e9-8ece-410f6cce5e4e.png)

**Comment by jorisvergeer** [comment]
@stevenbrix I've done that, I have 3 files totalling about 2.5GiB of data. How do you want to receive it?

**Comment by stevenbrix** [comment]
If you can upload it to some cloud storage provider like OneDrive or DropBox and provide me a link so I can download them that would be great! Or if you'd prefer, I could provide a link to my OneDrive for you to upload them to.

**Comment by jorisvergeer** [comment]
@stevenbrix 
Here you go, I've uploaded it temporary to my own VPS, tell me when you are done.

https://vpn01.jorisvergeer.nl/wpftest/

**Comment by stevenbrix** [comment]
@jorisvergeer got it, thank you!!

**Comment by jorisvergeer** [comment]
@stevenbrix I see that you requested both core and framework. I only did a run with core. Let me know if you want data for framework also still.

**Comment by stevenbrix** [comment]
@jorisvergeer that would be super helpful! It'll be helpful to compare the framework and core traces from the same machine.

**Comment by jorisvergeer** [comment]
Just a sec.

Meanwhile:
Behaviour also appears in the visual studio editor:

![image](https://user-images.githubusercontent.com/1906639/63546934-3aaa0f80-c52b-11e9-9330-b9f7fac68e0a.png)

**Comment by jorisvergeer** [comment]
@stevenbrix I did a run win .net 4.7.2, no weird behaviour as expected. 

In about 14 minutes there will be a file WpfTest01_net472.zip in the same location.
I guess that file will be there by the time you see this.

**Comment by vatsan-madhavan** [comment]
> If you could try and build the D3D9 sample [here](https://github.com/microsoft/Windows-classic-samples/tree/master/Samples/Win7Samples/multimedia/direct3d/direct3d9ex) and see if there are any issues with it. 

Did this run without problems ?

**Comment by PascalKoe** [comment]
I don't know why, but I can't reproduce the bug anymore...
The only thing I can remeber I changed are my two displays (Dell U2412M 16:10 1920x120 and an old Samsung 1280x1024) to one 32:10 Samsung display.
Just tried with the old monitors but no luck so far. Going to take a spare drive and do a fresh install of Windows to see if it then reproduces again...

@stevenbrix You asked for the specs of the custom build computer:
CPU: Intel Core i9-9900k
Graphics: Intel(R) UHD Graphics 630
Motherboard: Asus ROG STRIX Z390-F GAMING
RAM: HyperX Predator DDR4-3200 32GB Kit
SSD: Samsung 970 Evo M.2 1000GB

**Comment by jorisvergeer** [comment]
> > If you could try and build the D3D9 sample [here](https://github.com/microsoft/Windows-classic-samples/tree/master/Samples/Win7Samples/multimedia/direct3d/direct3d9ex) and see if there are any issues with it.
> 
> Did this run without problems ?

I am trying now

**Comment by jorisvergeer** [comment]
@stevenbrix @vatsan-madhavan 
I do run into issues. 

When started with "Intel integrated graphics":
I could increase complexity until i became bored
![image](https://user-images.githubusercontent.com/1906639/63551634-f1ab8880-c535-11e9-8009-d9f7acdcc3d1.png)

When started with "High Performance NVIDIA processor":
Initially a lot of black.
![image](https://user-images.githubusercontent.com/1906639/63551838-75657500-c536-11e9-95d1-d425d397e5a4.png)

When playing with increase and decrease complexity, from time to time the background comes back to live, only to show just a couple of frames, or be stuck when changing complexity.

![image](https://user-images.githubusercontent.com/1906639/63552030-f4f34400-c536-11e9-9c2e-a1ad0a40d678.png)

**Comment by jorisvergeer** [comment]
Note that with WPF it does not matter if I start my example app with one or the other graphics processor. Both show the same glitches.

**Comment by stevenbrix** [comment]
@jorisvergeer can you try disabling NahimicOSD and see if that fixes the issue?

**Comment by jorisvergeer** [comment]
@stevenbrix Which parts of the code do you want me to comment out?

**Comment by jorisvergeer** [comment]
@stevenbrix I think you are on to something

I stopped the Nahimic Service, and suddenly WPF acted just fine.

**Comment by jorisvergeer** [comment]
@stevenbrix I started the service again, and the glitches were back

**Comment by stevenbrix** [comment]
@jorisvergeer sounds like we've definitely found our culprit then! I have no idea why that audio service would be causing this. I've sent them a support request, hopefully they are responsive 🤞

**Comment by jenatali** [comment]
@jorisvergeer @stevenbrix my best guess here is that Nahimic has an overlay that they can render. When they do their overlay rendering, they probably change the viewport to cover the entire swapchain, and don't change it back to what the application had set. Then when WPF goes to render, it assumes that the viewport is still what it was last time, and if the only thing that changed was the same thing that changed last time, then it doesn't try to update the viewport. If Nahimic was involved, then it ends up rendering something that was supposed to be a sub-rect into the entire swapchain.

In this case, they're not drawing their overlay... but they're still changing the viewport it seems. This is more or less a guess, but it fits the symptoms and wouldn't surprise me.

**Comment by jorisvergeer** [comment]
@stevenbrix @jenatali @vatsan-madhavan 
If @jenatali his hypothesis is true, is it then not a bug in WPF that it assumes that the viewport was not changed. Can we not just always set the correct viewport?

**Comment by weltkante** [comment]
@jorisvergeer no its not a bug in WPF if some 3rd party application hijacks Windows APIs and changes their semantics. The viewport is a variable of the DirectX session and as such under the control of WPF. If this were C# this would be the equivalent of someone else going around changing random private variables via reflection.

**Comment by jorisvergeer** [comment]
@weltkante That could be, but somehow from all applications I am running, only WPF has this issue. 

And also, I just won't use WPF in my applications if I know that it WILL glitch on fresh out of the box computers from some brands.

**Comment by weltkante** [comment]
Adding workarounds in WPF will just make the 3rd party product staying broken and failing for any number of other applications. You are fixing your immediate problem by making it worse for everyone else in the long term. There needs to be back pressure to vendors and distributors of broken "improvement" hacks to actually fix or remove them. This can only happen by taking the time to identify them as broken. Since the symptoms have been root caused once now, future identification will probably be a lot easier.

I vote for not adding workarounds to correct code. If this is so bad to warrant logic in WPF it should be checking the modules list and blacklisting known bad modules, like it has been practice with video drivers containing known bugs for a long time.

The decision is on Microsoft of course.

**Comment by PudottaPommin** [comment]
Just add to the possible culprits, I did not have Nahimic, but Realtek Sonic Studio, which caused this buggy behaviour :)

**Comment by arpitmathur** [comment]
Closing this issue since this is externally caused.

**Comment by jorisvergeer** [comment]
I don't agree that much. External software might be triggering this, but if WPF core is the only library that has issues, then there is something ~wrong~ different with WPF core.

I sure as heck won't adopt WPF core in production applications knowing that it glitches with at least 2 different pieces of software that comes with fresh PC's.

**Comment by zamoldar** [comment]
the issue happing on window mode, works fine in AllowsTransparency="True" WindowStyle="None".

**Comment by fatihyildizhan** [comment]
#2299 This happened to me with .Net  Core 3.1 WPF application.  I am not using Nahimic and Realtek. I couldn't fix it so, I created a project with .Net Framework 4.8 WPF application and the problem goes away.

**Comment by sunmachine** [comment]
Chiming in - still happening as of dotnet core 3.1.100.

**Comment by zamoldar** [comment]
borless window(allow transparency) fixing the issue

**Comment by sunmachine** [comment]
It's a shame I can't get the default window behaviour to work, though. For now, since I'm just working with toys, this is fine. Thank you @zamoldar .

**Comment by fatihyildizhan** [comment]
I started to a new production project with .Net Core 3.1 because it has LTS support then I had to switch the .NET Framework due to this UI problem.

**Comment by jorisvergeer** [comment]
I can conform that my problems (which I still have) are going away when I set the `WindowStyle` to `"None"` and `AllowsTransparency` to `"True"`

as mentioned by @zamoldar

**Comment by johanli-axis** [comment]
We've seen users with this issue in our application and we're using .NET Framework, so the bug is not exclusive to .NET Core

**Comment by pcfulife** [comment]
Nahimic was updated to v1.4.1.0 which fixes this bug! yeah!

**Comment by rladuca** [comment]
@pcfulife Thanks so much for updating this issue, that's great!

Out of curiosity, did you see any release notes for the update?  My short searching for them wasn't successful.  I'd just like to see if they explicitly mention this issue just in case they only accidentally fixed it.  I wouldn't want the behavior to break again by happenstance.  I'll keep looking later on, but if you happened to have them handy (or even one of there installers that shows the notes) that would be helpful!

**Comment by pcfulife** [comment]
@rladuca I found release note at Microsoft Store.

> What's new on Nahimic 1.4.1?
> - Fix on **software compatibility**, incl. League of Legends fix
> - New supported models

There is no explict mention of WPF. But, they said that some compatibility issue was fixed.

**Comment by rladuca** [comment]
@pcfulife Thanks for taking a look!  A little more vague than I would like unfortunately.  I'll see if we can get more details out of this.

**Comment by jorisvergeer** [comment]
@pcfulife I saw that there inded were updates for the drivers. Unfortnately, those updates were not installed automatically. I had to go to drivers and updated them under "software components" manually.

At first glance, some of the applications that had glitches look OK now. (At least in the first minute of usage)

**Comment by benjamincoven** [comment]
Hello everyone,

I have the same problem, but I didn't find any nahimic driver on my computer. Is there anyway to check if nahimic is installed and which version? If Nahimic is not installed onmy computer, should I open another issue? Note that I found some scenarios in my app when the problem happens and some it does not on the same controls.

**Comment by realflight1** [comment]
For anyone with ASUS ROG machines, the Sonic Suite application that comes with the system causes this issue (I assume it uses Nahimic internally).

**Comment by roceh** [comment]
I agree with Microsofts approach as this overlay software is basically malware in my eyes and causes problems with a lot of apps. But it is quite widespread unforunately and only the latest releases of their software will have the .net core detection fixes in, so I've found a couple of workarounds. 

One is to inspect the modules at startup (Process.GetCurrentProcess().Modules) and if a module named "NahimicOSD.dll" is there it will downgrade to software rendering (RenderOptions.ProcessRenderMode = RenderMode.SoftwareOnly). At least the app will run without corruption and on other PCs you will still get acceleration.

The second method which keeps the hardware accleration is to put a d3d9.dll proxy in your application directory, this seems to stop the overlay injectors. I took this one https://github.com/maltsevda/FakeD3D9 and removed the adjustments specific for that use case and made it a straight proxy https://github.com/roceh/FakeD3D9 .  It seems to stop Nahimic/Sonic Suite that gets installed with my realtek audio drivers for my motherboard (ASUS ROG 370F).

**Comment by sirJimmy1** [comment]
Hi, still happening in .net5. Just migrate my wpf app from .Net framework 4.6 where it was working fine and discovered this issue.

**Comment by weltkante** [comment]
@sirJimmy1 note that the behavior was not caused by WPF, but by faulty 3rd party software injecting itself incorrectly into the .NET Core program. Updating to .NET 5 will likely not change how the 3rd party software behaves.

**Comment by sirJimmy1** [comment]
@weltkante my bad.... I am blind. Thank you. 
Updating Nahimic now so will see if it will fix it.

**Comment by konradsikorski** [comment]
i have the same problem described in #4141 
Today was trying to find the app responsible for this and found it. In my case it was "**Asus Rog Phoebus Sonic studio** v1.1.14". After killing it from system try the problem is gone now.

**Comment by Squall-Leonhart** [comment]
This issue isn't specific to .net core, it affects any system with .net 4.8 installed even if the application was targetted to 4.7.2 or earlier.

**Comment by HitEmUp** [comment]
Same issue, all  buttons like the Visual Studio Installer (bottom right) have this problem.
I tried update my Nvidia drivers... windows 10 is up to date. 

![Windows10-ButtonBug](https://user-images.githubusercontent.com/10390619/126177259-d1446068-bfc1-478d-91a3-c20bca79d8a3.png)

**Comment by jorisvergeer** [comment]
@HitEmUp Like everyone else here. It is probably caused by some crapware that came with your sound drivers that likes to inject itself into various applications not on it's blacklist that make use of DirectX for drawing. Probably for some 3d spatial aware sound effect or something, or for injecting an overlay of some sort. (Guess what, all new releases of dotnet core are probably not prematurely listed in its blacklist)

**Comment by HitEmUp** [comment]
@jorisvergeer thx for your hint, i did check some applications for overlays and turn everything off i found.
But still no fix, do u mean i need to update my sound drivers?
What did help you to fix it? 
It's windows wide, means, its everywhere (WPF / UWP Apps), .Net 5 . Net 4.8, Core .Net 3.1... doesn't matter. Looks like a Windows wide bug to me.

**Comment by HitEmUp** [comment]
> @jorisvergeer @stevenbrix my best guess here is that Nahimic has an overlay that they can render. When they do their overlay rendering, they probably change the viewport to cover the entire swapchain, and don't change it back to what the application had set. Then when WPF goes to render, it assumes that the viewport is still what it was last time, and if the only thing that changed was the same thing that changed last time, then it doesn't try to update the viewport. If Nahimic was involved, then it ends up rendering something that was supposed to be a sub-rect into the entire swapchain.
> 
> In this case, they're not drawing their overlay... but they're still changing the viewport it seems. This is more or less a guess, but it fits the symptoms and wouldn't surprise me.

Nvm. it was NahimicService!!! Turned it off, nice, thanks guys.

**Comment by Squall-Leonhart** [comment]
@jorisvergeer the software might be crapware, but it was previously not causing this prior to recent windows 10 cumulative updates unless a severely out dated version was installed.

Like mentioned in other posts here Nahimic did introduce measures to prevent it from causing this, and these measures have been rendered ineffective as a result of some change in the OS.

**Comment by Daniel15** [comment]
I can't believe that two years later, this garbage Nahimic stuff still has the same issue. I read online that it comes with something called "Sonic Suite", which was apparently bundled with the sound drivers for my motherboard. I uninstalled that as well as some other Sonic thing, but they left the Nahimic Service in place. I couldn't figure out how to properly uninstall the service so I first disabled it and then deleted the `C:\Windows\system32\NahimicService.exe` file (protip: good, well-written apps don't just drop their executables into system32). I'm worried a driver update will bring it back, in which case I'll just put a dummy EXE there and change the permissions so that nobody other than some random user account on my PC can write to the file.

**Comment by zyo2012** [comment]
I know this topic is old but on a brand new computer on windows 10 I was getting that issue and it was cause by the Asus Motherboard tools. I've uninstall everything that had the word "Asus" and that fixed my "mouse over" problem.

**Comment by Xinnony** [comment]
Same here for all WPF program, i uninstall all Sonic app (ASUS), and work perfectly after reboot (NahimicService is run).

**Comment by anose001** [comment]
Thanks! The first suggestion worked and I was able to confirm that Nahimic loads into the modules that startup. Although I would like to try the second one as well but was not able to find a dll file. Do I need to compile it after downloading the github repo?

Also, this is a little strange, but for over a year everything was perfectly fine and then all of a sudden this started occurring. I am wondering whether the audio driver update or anything... that would be quite unfortunate.

**Comment by walterlv** [comment]
I collected the modules of a broken WPF app, and find three unexpected modules:

1. C:\ProgramData\AWHeadset\DellInk.AlienwareSoundCenter\Modules\ScheduledModules\DellInc.AlienwareSoundCenterDevProps2.dll
2. C:\ProgramData\AWHeadset\DellInk.AlienwareSoundCenter\Modules\ScheduledModules\NahimicOSD.dll
3. C:\ProgramData\AWHeadset\DellInk.AlienwareSoundCenter\Modules\ScheduledModules\ProductInfo.dll

Yes! It's the NahimicOSD.dll that is mentioned by most of you, and the AlienwareSoundCenter does this!

**Comment by anose001** [comment]
@walterlv, are you suggesting to remove those?

**Comment by walterlv** [comment]
@anose001 Yes.

I've removed the `ScheduledModules` folder and everything goes back to fine.

**Comment by anose001** [comment]
@walterlv , that worked. Curious to see what comes out and learn more. Thanks and cheers

**Comment by TrashcanCoder** [comment]
Deleting 
C:\ProgramData\AWHeadset\DellInc.AlienwareSoundCenter\Modules\ScheduledModules\NahimicOSD.dll
solved the Problem.
(I have to kill some Processes that blocked the remove for the .dll)

No Luck:
Reinstall Windows because AlienwareSoundCenter was installed automatically after plug in Headset.

**Comment by walterlv** [comment]
@TrashcanCoder If you don't want to find and kill the processes, you can try to rename the dll.

**Comment by gmgunderground** [comment]
Hi, same problem on my Alienware Area 51m R2

**Comment by walterlv** [comment]
> Hi, same problem on my Alienware Area 51m R2
> 
> Renaming NahimicOSD.dll no more problems, but what is the purpose of injecting NahimicOSD ?

NahimicOSD is an on-screen display library that displays something on any software using DirectX as its rendering engine. Unfortunately, the additional rendering blooms the rendering of WPF apps (DirectX 9 application precisely).

You can see more details on my post:

* [All WPF apps are rendered in a mess!](http://blog.walterlv.com/post/wpf-renders-wrong-because-of-nahimicosd-en.html)

**Comment by TrashcanCoder** [comment]
**Updated to Win11**

Same Error again:
Deleting NahimicOSD.dll did not work; Some processes accesses the file (Unable to kill).
C:\ProgramData\AWHeadset\DellInc.AlienwareSoundCenter\Modules\ScheduledModules\x64\NahimicOSD.dll

But with file -> Properties -> Security: Deny all access works perfect.

**Comment by rephiled** [comment]
I have this issue on a new Alienware Aurora 13 I cant find anything like Nahimic on it and I don't have any computer skill what can I do

**Comment by Squall-Leonhart** [comment]
Nahimic is on several alienware machines, 

https://www.reddit.com/r/MSI_Gaming/comments/l45fnj/guide_how_to_uninstall_nahimic_completely_from/

**Comment by rephiled** [comment]
Thanks for this

**Comment by rephiled** [comment]
> Nahimic is on several alienware machines,
> 
> https://www.reddit.com/r/MSI_Gaming/comments/l45fnj/guide_how_to_uninstall_nahimic_completely_from/
Thanks for this

**Comment by oli-g-sk** [comment]
> I have this issue on a new Alienware Aurora 13 I cant find anything like Nahimic on it and I don't have any computer skill what can I do

1. Open **Start** and search for **Services** 
2. Look for the _Nahimic service_, **right click** and choose **Properties**
3. Set its Startup Type to **Disabled**
4. Restart

### 🖼️ Screenshots
1.  https://ibb.co/S32LbVm
2. https://ibb.co/DtGwSY7
3. https://ibb.co/sjPtKJ2

**Comment by Squall-Leonhart** [comment]
This is reportedly resolved with standalone nahimic 1.8.6, and vendor softwares including it.
